rpc/http.go

```diff
@@ -254,7 +254,17 @@ type httpServerConn struct {
 func newHTTPServerConn(r *http.Request, w http.ResponseWriter) ServerCodec {
 	body := io.LimitReader(r.Body, maxRequestContentLength)
 	conn := &httpServerConn{Reader: body, Writer: w, r: r}
-	return NewCodec(conn)
+	enc := json.NewEncoder(conn)
+	dec := json.NewDecoder(conn)
+	dec.UseNumber()
+	encoder := func(v any) error {
+		err := enc.Encode(v)
+		if f, ok := w.(http.Flusher); ok {
```

Comment:
Need to add a comment here, something like

```
// In case of a timeout error, the response must be written before the HTTP server's write
// timeout occurs. So we need to flush the response here.
```